x-pack/plugin/logstash/src/test/java/org/elasticsearch/xpack/logstash/PipelineRequestResponseSerializationTests.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.logstash;
+
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.logstash.action.DeletePipelineRequest;
+import org.elasticsearch.xpack.logstash.action.DeletePipelineResponse;
+import org.elasticsearch.xpack.logstash.action.GetPipelineRequest;
+import org.elasticsearch.xpack.logstash.action.GetPipelineResponse;
+import org.elasticsearch.xpack.logstash.action.PutPipelineRequest;
+import org.elasticsearch.xpack.logstash.action.PutPipelineResponse;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class PipelineRequestResponseSerializationTests extends ESTestCase {
```

Comment:
Weak recommendation: Break these out into `AbstractWireSerializingTestCase`s and get equals/hashcode tests for free, plus make it a bit easier to find the tests for each class.

Reply:
I've done this.